mlflow/gateway/schemas/chat.py

```diff
@@ -55,7 +55,7 @@ class CandidateMetadata(ResponseModel):
 
 class Candidate(ResponseModel):
     message: ResponseMessage
-    metadata: CandidateMetadata
+    metadata: Optional[Dict[str, str]]
```

Comment:
Metadata will likely contain values that aren't just strings. Can we keep the `CandidateMetadata` class and instead update the model to `extra.Allow` so that additional fields beyond `finish_reason` are supported?

Reply:
great idea. Added to class constructor for CandidateMetadata